cabot/cabotapp/run_window.py

```diff
@@ -0,0 +1,96 @@
+import json
+from datetime import time
+from dateutil import rrule
+from django.utils import timezone
+
+
+class CheckRunWindow:
+    class Window:
+        start_time = None  # type: time
+        end_time = None  # type: time
+        recurrence = None  # type: rrule.rrule
+
+        def __init__(self, start_time, end_time, recurrence):
+            # type: (time, time, rrule.rrule) -> None
+            assert isinstance(start_time, time)
+            assert isinstance(end_time, time)
+            assert isinstance(recurrence, rrule.rrule)
+            self.start_time = start_time
+            self.end_time = end_time
+            self.recurrence = recurrence
+
+        def active(self, now=timezone.now()):
+            # type: (timezone.datetime) -> bool
+            assert now.tzinfo == timezone.utc
+
+            dtstart = (now.replace(hour=self.start_time.hour, minute=self.start_time.minute, second=0, microsecond=0)
+                       - timezone.timedelta(days=1))
```

Comment:
Why are you subtracting a day? Can you comment about what this is doing bc I'm confused

Reply:
Added a comment to the code that should explain, lmk if still not clear.